FloripaTrans/FTRouteDetailTableViewDelegateHandler.m

```diff
@@ -0,0 +1,237 @@
+//
+//  FTRouteDetailTableViewDelegateHandler.m
+//  FloripaTrans
+//
+//  Created by Vítor Makoto Matayoshi de Moraes on 3/7/15.
+//  Copyright (c) 2015 Vitor Makoto. All rights reserved.
+//
+
+#import "FTRouteDetailTableViewDelegateHandler.h"
+#import "FTDeparture.h"
+#import "FTStop.h"
+
+
+@interface FTRouteDetailTableViewDelegateHandler()
+@property (strong, nonatomic) NSArray *streets;
+
+@property (strong, nonatomic) NSMutableArray *weekdayDepartures;
+@property (strong, nonatomic) NSMutableArray *saturdayDepartures;
+@property (strong, nonatomic) NSMutableArray *sundayDepartures;
+@end
+
+@implementation FTRouteDetailTableViewDelegateHandler
+
+- (instancetype)init {
+    
+    if(self = [super init])
+    {
+        self.streets = [NSArray array];
+        
+        self.weekdayDepartures  = [NSMutableArray array];
+        self.saturdayDepartures = [NSMutableArray array];
+        self.sundayDepartures   = [NSMutableArray array];
+    }
+    
+    return self;
+}
+
+/**
+ *  Method that updates the streets for the streets array data source and refresh the tableview
+ *
+ *  @param streets streets
+ */
+- (void)updateStreets:(NSArray *)streets {
+    self.streets = streets;
+    if(self.delegate && [self.delegate respondsToSelector:@selector(refreshTableViewSection:)]) {
+        [self.delegate refreshTableViewSection:0];
+    }
+}
+
+/**
+ *  Method that updates the departures arrays data source and refresh the tableview
+ *
+ *  @param departures departures
+ */
+- (void)updateDepartures:(NSArray *)departures {
+    
+    self.weekdayDepartures  = [NSMutableArray array];
+    self.saturdayDepartures = [NSMutableArray array];
+    self.sundayDepartures   = [NSMutableArray array];
+    
+    for(FTDeparture *d in departures) {
+        if(d.calendar == WEEKDAY){
```

Comment:
My bad on this one. Because I was checking string in the other nested ifs, I accidentally did in this one. I'm going to replace it with a switch right away. Thanks.